manuscript/manuscript.Rmd

```diff
@@ -242,39 +238,84 @@ When the agent cannot fully observe the state of the system, reinforcement learn
+Yet, for many decision-making problems, the state-action space cannot be tractably represented in a table; for this case, one effective alternative is to use a neural network to approximate the Q-function, a method called deep Q-learning. 
+The general process for deep Q-learning differs from policy gradient methods, since instead of trying to maximize the return directly, the objective is to minimize the Bellman error.
+The Bellman error is a measure of how distant a given value function is from that value function which has been re-evaluated with recent environment interaction.
+Suppose we attempt to approximate the Q-function with a neural network and thus represent this approximate Q-function as as $Q_\phi$, the Bellman error for one state-action pair would be
 
 $$
-  A^\pi(s_t, a_t) = Q^\pi(s_t, a_t) - V^\pi(s_t)
-$$ 
-Learning $Q^\pi$, commonly called the Q-value function, is more powerful than learning the state-value function as Q-values easily prescribe what action to take when in a certain state, $a_t^* = \underset{a_t}{\text{argmax}} \, Q^\pi (s_t, a_t)$.
-Conversely, if we know the state-value function, we would have to perform a maximization over an expected value to find the optimal action, $a_t^* = \underset{a_t}{\text{argmax}} \, \mathbb{E}_{s_{t+1} \sim \,T(s_{t+1} \mid s_t, a_t)}[V^\pi(s_{t+1})]$, which is less tractable.
-Yet, learning the advantage function is often preferred in practice as advantage estimators favorably have lower variance than state-value and Q-value estimators [cite schulman gae].
+  \mathcal{E} = r(s_t, a_t) + \gamma \, \underset{a}{\text{max}} \, Q_\phi(s_{t+1}, a) - Q_\phi(s_t, a_t).
+$$
+
+See the appendix/reference for details on how the Bellman Error is derived from eq. [@CITE]. 
+The general scheme for approximate value-learning with deep neural networks is to sample trajectories from the environment, evaluate the Bellman Error across these samples -- with the samples indexed by $i$ --, and then perform the following gradient descent:
+
+$$
+  \phi_{t+1} = \phi_t - \alpha \nabla_\phi \, \sum_i \mathcal{E}_i^2 
+$$
+
+When being evaluated, a deep Q-learning agent will select actions by inputting the current state to the Q network and then identifying the state-action pair with the highest estimated Q value.
+The benefit of using neural networks to approximate Q-functions is that neural networks are able to generalize Q functions over unseen state-action pairs, whereas classic methods like TD-learning are not able to generalize [@CITE].
+But there a host of problems with getting neural-network-based value-learning agents to work in practice; see @CITE for more insight on these issues.   
+The primary concern with these methods is that they tend to suffer from bias which is introduced by fitting a value network to a target that incorporates an estimated value function [@CITE].  
+
+## Actor-Critic
```

Comment:
I think instead of making the header 'Actor Critic' specifically here, we want to acknowledge that there's a very wide and growing array of 'tricks' that modify the basic approaches, and that most algorithms are doing combinations of these.  we can use actor-critic as such a nice illustration of that (e.g. pointing out that rather than do one or the other of the two, most modern algorithms do both using an 'actor' and a 'critic') but without going into any detail.  We also want to nod to there being a lot more that we simply don't touch on here, stuff like action clipping, delayed experience replay, etc etc.